app/src/main/java/com/hfad/sapper/adapter/CellAdapter.kt

```diff
@@ -0,0 +1,115 @@
+package com.hfad.sapper.adapter
+
+import android.graphics.Color
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.recyclerview.widget.RecyclerView
+import com.hfad.sapper.R
+import com.hfad.sapper.data.Cell
+import com.hfad.sapper.ui.main.CLEAR_CELL
+import com.hfad.sapper.ui.main.FIELD_SIZE
+import com.hfad.sapper.ui.main.MINE_CELL
+import com.hfad.sapper.ui.main.TRIGGER_MINE_CELL
+
+class CellAdapter(private var playingField: MutableList<Cell> = mutableListOf()) :
+    RecyclerView.Adapter<CellAdapter.ViewHolder>() {
+    private var itemListener: ItemClickListener? = null
+
+    fun setListener(itemClickListener: ItemClickListener) {
+        this.itemListener = itemClickListener
+    }
+
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(viewGroup.context)
+            .inflate(R.layout.item_cell, viewGroup, false)
+
+        return ViewHolder(view)
+    }
+
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        val cell = playingField[position]
+        when (cell.valueCell) {
+            CLEAR_CELL, MINE_CELL -> {
+                viewHolder.cellButton.setBackgroundColor(Color.GRAY)
```

Comment:
Here, when all RecyclerView elements are visible, it's not a problem not to update text. But if you display the field as a scrollable list, you could face previously set text. So, set correct text value for each field.